stylix/autoload.nix

```
@@ -8,21 +8,37 @@ builtins.concatLists (
     path: kind:
     let
       file = "${inputs.self}/modules/${path}/${for}.nix";
-      module = import file;
-      # Assume that a function-module whose first arg doesn't have any de-
-      # structured attrs is actually expecting `mkTarget` as its first arg;
-      #
-      # `mkTarget` cannot be distribtued normally through the module system
-      # due to issues of infinite recursion
-      expectsMkTarget =
-        builtins.isFunction module && builtins.functionArgs module == { };
+      value = import file;
+
+      # Detect whether the file's value has an argument named `mkTarget`
+      useMkTarget =
+        builtins.isFunction value && (builtins.functionArgs value) ? mkTarget;
+
+      # NOTE: `mkTarget` cannot be distributed normally through the module system
+      # due to issues of infinite recursion.
+      mkTarget = import ./mk-target.nix;
     in
     lib.optional (kind == "directory" && builtins.pathExists file) (
-      if expectsMkTarget then
+      if useMkTarget then
+        { config, ... }@args:
+        let
+          # Based on `lib.modules.applyModuleArgs`
```

```suggestion
          # Based on `lib.modules.applyModuleArgs`
          #
          # Apply `mkTarget` as a special arg without actually using `specialArgs`,
          # which cannot be defined from within a configuration.
```

Maybe we should be more explicit about _why_ we do this?